File: SAMPLES/CPP/test_binexpr.cpp
// test_binexpr.cpp - Test TODO 12 : expressions binaires C++
// Arithmetique, bits, comparaisons, logique, ternaire, virgule

int main() {
    // --- Arithmetique entiere ---
    int a = 10 + 3;
    int b = 10 - 3;
    int c = 4 * 5;
    int d = 20 / 4;
    int e = 17 % 5;

    // --- Operateurs bit a bit ---
    int f = 0xFF & 0x0F;
    int g = 0xF0 | 0x0F;
    int h = 0xFF ^ 0x0F;
    int i = 1 << 4;
    int j = 32 >> 2;

    // --- Comparaison ---
    int k = 5 == 5;
    int l = 5 != 3;
    int m = 3 < 5;
    int n = 5 <= 5;
    int o = 7 > 3;
    int p = 3 >= 3;

    // --- Logique court-circuit ---
    int q = 1 && 1;
    int r = 1 && 0;
    int s = 0 || 1;
    int t = 0 || 0;

    // --- Ternaire ---
    int u = 1 ? 42 : 99;
    int v = 0 ? 42 : 99;

    // --- Precedence mixte ---
    int w = 2 + 3 * 4;
    int x = (2 + 3) * 4;

    // --- Variables et operations ---
    int y = a + b;
    int z = y * 2 - a;

    return 0;
}

File: SAMPLES/PYTHON/test_binexpr.py
# test_binexpr.py - Test TODO 11: EXPRESSIONS BINAIRES
# Compile : PYTHON86 test_binexpr.py test_binexpr.asm

# Arithmetique entiere : +, -, *, //, %
a = 10 + 5
b = 10 - 3
c = 4 * 7
d = 20 // 3
e = 17 % 5

# Puissance **
f = 2 ** 10
g = 3 ** 0

# Operateurs bit a bit : &, |, ^, <<, >>
h = 0xFF & 0x0F
i = 0xF0 | 0x0F
j = 0xFF ^ 0x0F
k = 1 << 8
l = 256 >> 4

# Comparaisons entieres : ==, !=, <, <=, >, >=
m = 10 == 10
n = 10 != 5
o = 3 < 7
p = 7 <= 7
q = 9 > 2
r = 5 >= 5

# Logique court-circuit : and, or
s = True and True
t = True and False
u = False or True
v = False or False

# Logique not (unaire)
w = not False
x = not True

# Concatenation de chaines
s1 = "hello" + " " + "world"

# Repetition de chaine
s2 = "ab" * 3

# Comparaison de chaines
c1 = "abc" == "abc"
c2 = "abc" != "def"
c3 = "abc" < "def"
c4 = "xyz" > "abc"
c5 = "abc" <= "abc"
c6 = "abc" >= "abc"

# Operateur in / not in (listes)
r1 = 1 in [1, 2, 3]
r2 = 5 not in [1, 2, 3]

# Identite is / is not
r3 = None is None
r4 = 42 is not None

# Combinaisons de precedence
z1 = 2 + 3 * 4
z2 = (2 + 3) * 4
z3 = 2 ** 3 ** 2
z4 = 10 - 3 * 2 + 1

# Expressions avec variables
x1 = 10
x2 = 20
x3 = x1 + x2
x4 = x1 * x2
x5 = x2 // x1
x6 = x1 == x2
x7 = x1 < x2
